File: web/src/utils/isEscaped.ts
import { BACKSLASH_CODE } from "./charCodes";

export default function isEscaped(text: string, i: number): boolean {
	return text.charCodeAt(i - 1) === BACKSLASH_CODE && text.charCodeAt(i - 2) !== BACKSLASH_CODE;
}

File: zig/src/utils/isEscaped.zig
pub fn isEscaped(text: []const u8, i: usize) bool {
    if (i == 0) return false;
    if (text[i - 1] != '\\') return false;
    if (i == 1) return true;
    return text[i - 2] != '\\';
}
